tests/capture/transforms/test_capture_graph_decompose.py

```diff
@@ -271,3 +271,57 @@ def f(x, y, z):
             qml.RY(qml.math.array(-0.2, like="jax"), wires=[0]),
             qml.RX(qml.math.array(-0.1, like="jax"), wires=[0]),
         ]
+
+    @pytest.mark.integration
+    def test_cond(self):
+        """Tests that a circuit containing conditionals can be decomposed correctly."""
+
+        @DecomposeInterpreter(gate_set={"CNOT", "RX", "RY", "RZ"})
+        def f(x, wires):
+
+            def true_fn():
+                qml.CRX(x, wires=wires)
+
+            def false_fn():
+                qml.CRZ(x, wires=wires)
+
+            qml.cond(x > 0.5, true_fn, false_fn)()
+
+        # The PLxPR is constructed with the true_fn branch
+        jaxpr = jax.make_jaxpr(f)(0.6, [0, 1])
+        collector = CollectOpsandMeas()
+        # Tests that the false_fn branch is also decomposed
+        collector.eval(jaxpr.jaxpr, jaxpr.consts, 0.2, *[0, 1])
+        assert collector.state["ops"] == [
+            qml.RZ(qml.math.array(0.1, like="jax"), wires=[1]),
+            qml.CNOT(wires=[0, 1]),
+            qml.RZ(qml.math.array(-0.1, like="jax"), wires=[1]),
+            qml.CNOT(wires=[0, 1]),
+        ]
```

Comment:
This test is not checking the true branch at all. It seems worthwhile to add that to me.